website/blog/2024-08-20-introducing-fe-libs-tailwind.md

```diff
@@ -0,0 +1,161 @@
+
+## Benefits of the new setup
+
+Here are the stats after testing the setup:
+
+- sites are **5x** times faster on the frontend and **3x** times faster on the backend.
+- load time is absolutely **minimal**
+- **50%** fewer DOM elements
+- build time is **10x+** times faster.
+- final bundle sizes for JS and CSS are **3x** smaller.
+
+Additional benefit is that there are a lot of Tailwind community sites with ready-made components, which allow you to copy the component and modify it.
+
+## Initial impressions from the team
+
+_“The project I’m working on is the first one where we started using FE Libs Tailwind. To be honest, I was quite worried about working on a new project with this setup, as the estimate was a bit lower than what we usually estimate._
+
+_After a few initial hiccups with the setup, I found it was much easier to work with and it offered a lot of possibilities. I was really surprised how easy it was to set up the default values like font sizes and colors._
+
+_Using Tailwind classes speeded up development process quite significantly and I managed to implement all blocks from the initial design phase one week before the deadline.”_
+
+## Requirements and how to install it
+
+FE Libs Tailwind requires at least **PHP 8.3**. To install it, run the following command in your project folder:
+
+```json
+npx eightshift-create theme
+```
+
+The project setup script is now also upgraded and it offers a choice between standard FE Libs and FE Libs Tailwind. If you choose *simple setup*, it installs FE Libs Tailwind.
+
+## Project structure
+
+The project structure is quite similar to the standard Eightshift setup, but there are a few things to take note of.
+
+`tailwind.config.mjs` is the main config file for Tailwind where you define the colors, breakpoints, font sizes and all custom CSS classes you need to use in your project.
+
+When looking at the component or block structure, you may notice that most of them no longer have CSS files. As Tailwind offers lots of utility classes, in most cases you won’t need a CSS file for your component or a block. In case you need it, simply add `styles.css` or `styles-editor.css`. Notice how it’s no longer required to prefix the block/component name before the name of the CSS file.
+
+Tailwind classes are added in the `manifest.json` file of your block, but more on that in the next section.
+
+You will also notice that this setup uses **Eightshift UI Components** for rendering the options in the editor. We decided to decouple Eightshift UI Components from our setup to make maintenance easier. If you would like to know more, please check the [Eightshift UI Components documentation](https://eightshift.com/components/es-ui-components/getting-started).
```

Comment:
```suggestion
The project structure is quite similar to the standard Eightshift setup, but there are a few things to take note of.

`tailwind.config.mjs` is the main config file for Tailwind. There you define things like (project-specific) colors, font sizes, spacings, and any other Tailwind configuration options, or custom utilities you might need in your project. Breakpoints are still defined in the global manifest.

When looking at the component or block structure, you may notice that most of them no longer have stylesheets assigned. As Tailwind is utility class-based, in most cases you won’t need a CSS file for your component or a block. 
Note that you can also provide one-time arbitrary values in Tailwind (check the docs for details) if needed. In case you need it, simply add `styles.css`, `styles-editor.css`, or `styles-frontend.css`. Notice how it’s no longer required to prefix the block/component name before the name of the CSS file.

Tailwind classes can be shared between the frontend and the backend if added in the `manifest.json` file of your block or component, but more on that in the next section. You can always just write classes in the markup if needed.

You will also notice that this setup uses **Eightshift UI components** for rendering the options in the editor. We decided to decouple UI components from the setup to make maintenance easier. If you would like to know more, please check the [Eightshift UI components documentation](https://eightshift.com/components/es-ui-components/getting-started).
```